invenio_rdm_records/resources/serializers/csl/__init__.py

```diff
@@ -18,3 +28,92 @@ class CSLJSONSerializer(MarshmallowJSONSerializer):
     def __init__(self, **options):
         """Constructor."""
         super().__init__(schema_cls=CSLJSONSchema, **options)
+
+
+def get_citation_string(json, id, style, locale):
+    """Get the citation string from Citeproc."""
+    def _clean_result(text):
+        """Remove double spaces, punctuation and escapes apostrophes."""
+        text = re.sub(r'\s\s+', ' ', text)
+        text = re.sub(r'\.\.+', '.', text)
+        text = text.replace("'", "\\'")
+        return text
+
+    source = CiteProcJSON([json])
+    citation_style = CitationStylesStyle(validate=False, style=style,
+                                         locale=locale)
+    bib = CitationStylesBibliography(citation_style, source, formatter.plain)
+    citation = Citation([CitationItem(id)])
+    bib.register(citation)
+
+    return _clean_result(''.join(bib.bibliography()[0]))
```

Comment:
Yes, it should always return a list. Every time you call `register`, it will add the each `citation.cite` to the internal list of items, see [here](https://github.com/brechtm/citeproc-py/blob/9079882f5a5ca43be616dbe88c8e897c7374ed5b/citeproc/frontend.py#L85).

I think that since we only have 1 record, we always only one `cite`. No need to use `' '`, it will be always 1 element max.